src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
without this.coords will be failed tests here, because this.coords will be undefined and will not possible read field x or y in undefined.
better directly set coords with fields x, y
```suggestion
   this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
   };
```